src/storage.rs

```diff
@@ -3,12 +3,29 @@
 
 use crate::database::get_db_pool;
 use crate::didcomm_types::ForwardMsg;
+use cfg_if::cfg_if;
 
-pub struct Storage {
-    pool: sqlx::AnyPool,
+cfg_if! {
+    if #[cfg(feature = "AnyDB")] {
+        pub struct Storage {
+            pool: sqlx::AnyPool,
+        }
+    } else if #[cfg(feature = "PostgresqlDB")] {
+        pub struct Storage {
+            pool: sqlx::PgPool,
+        }
+    } else if #[cfg(feature = "MysqlDB")] {
+        pub struct Storage {
+            pool: sqlx::MySqlPool,
+        }
+    }
 }
```

Comment:
I don't think there's a need for the external crate. You could alternatively write this as 
```suggestion
pub struct Storage {
     #[cfg(feature = "any_db")]
     pool: sqlx::AnyPool
     #[cfg(feature = "postgres")]
     pool: sqlx::PgPool
     #[cfg(feature = "mysql")]
     pool: sqlx::MySqlPool
}
```

Reply:
I had tried to do this way before, but it would cause a `field already declared` error if more than one feature flag is enabled.  (Say by forgetting to disable default (Mysql) before activating another feature)

The cargo docs recommend cfg_if for cases where we have mutually exclusive / conflicting feature flags. 
https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features


With this code, when multiple are activated, the final backend will be selected in decreasing priority 
Any  > Postgres > Mysql  

Mysql is set as the default feature, so this also means users can just activate any of the other drivers without having to worry about calling with  `---no-default-features`

Reply:
That's precisely why it's better to have them conflict. Users should *not* have multiple backends selected at once, and this prevents that. The decreasing priority is a hidden side effect that users don't know about (yeah you could document it but it's still not going to be as straightforward as a compilation error).

Reply:
Alright. So I suppose a deliberate compilation error would make more sense. On detecting multiple features.

Reply:
They mention this example:
```
#[cfg(all(feature = "foo", feature = "bar"))]
compile_error!("feature \"foo\" and feature \"bar\" cannot be enabled at the same time");
```

Is there a way to say "when there's more than one of .." ? 
Am not able to find it here https://doc.rust-lang.org/reference/conditional-compilation.html#the-cfg-attribute

Reply:
I personally think so, yeah. Allowing a consumer to set multiple backends could be confusing. You can have some sanity checks that conditionally compile a [compile_error](https://doc.rust-lang.org/std/macro.compile_error.html) if two or more features are enabled to present a better error message rather than `field already declared`.

There are some operators on the `#[cfg()]` macro, such as `and`, `or`, `not` that can help with that.

Reply:
Thanks. This will make it, explicit.  I will look for those #[cfg()] macro operators. That should help.

Reply:
Seems like this is turning into an exercise of implementing `NXOR` gate on n=3 inputs, using[ (all, any, not)](https://doc.rust-lang.org/reference/conditional-compilation.html)   :smile:

Reply:
Yeah, my bad, it was `all`, `any` and `not`, not `and` and `or` 😅 . You don't have to do it all in one line though. 

Something like `#[cfg(all(feature = "feature_1", any(feature = "feature_2", feature = "feature_3")))]` and then just swapping them around on 3 different lines should be readable and would all annotate the same upcoming code block.

Reply:
Done! Emitting compilation error if more than one backend database feature flag selected. Also doing this using built in cfg predicates.

Reply:
Used this [cfg block](https://github.com/nain-F49FF806/axum-test-server/pull/6/files#diff-18c17f16d6477d10ed90360fe38e0e13fc117ecd9548f8f9374236c3ec7f8750R8-R15)

It works, and I think it's clearly readable, what's going on.